### PR TITLE
[logging] deprecation notices for SQLite

### PR DESCRIPTION
**Pull request by lilykuang**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Support for SQLite as a metadata database will be officially removed. SQLite may still be usable, but it will not be supported in future Alembic migrations. This PR is following step 1 on Immediate: plans from [SIP-33] https://github.com/apache/incubator-superset/issues/8874
- Add logging to indicate SQLite’s deprecation in the next version release
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@willbarrett

### CODE REVIEW

**Review by etr2460 — COMMENTED** [review]
I'd recommend using `logger.warn` throughout instead of `logger.info`, once you make the change though, it lgtm

**Comment by lilykuang** [comment]
Thank you @etr2460. It seems like logging.warn() has been deprecated for Python 3.4+, will use `logger.warning`

**Review by etr2460 — APPROVED** [review]
lgtm, didn't realize that `logger.warn` was deprecated. `logger.warning` is great!

File: superset/migrations/env.py [hunk 1]
```diff
@@ -34,10 +34,13 @@
 fileConfig(config.config_file_name)
 logger = logging.getLogger("alembic.env")
 
-
-config.set_main_option("sqlalchemy.url", current_app.config["SQLALCHEMY_DATABASE_URI"])
+DATABASE_URI = current_app.config["SQLALCHEMY_DATABASE_URI"]
+if "sqlite" in DATABASE_URI:
+    logger.warning("SQLite DB support may not be supported by in future version")
```
Comment (willbarrett) [inline superset/migrations/env.py:39]: Grammar nit: could we use "SQLite Database support for metadata databases will be removed in a future version of Superset."

**Review by willbarrett — APPROVED** [review]
LGTM aside from one grammar nit :)

**Review by villebro — APPROVED** [review]

**Review by craig-rueda — APPROVED** [review]

**Pull request merged**